mock/mock.go

```diff
@@ -578,6 +578,19 @@ func (m *Mock) IsMethodCallable(t TestingT, methodName string, arguments ...inte
 	return false
 }
 
+// NumberOfCalls returns the number of times a method was called
+func (m *Mock) NumberOfCalls(methodName string) int {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
```

Comment:
Would it not make sense to move this unlock to right after the critical section (i.e. between lines 590,591)? My thought is that it being explicitly there will help the critical section unnecessarily ballooning by accident.

Reply:
I'd ideally go with `defer Unlock` since it's more "failsafe" in case the function body panics, however in this case, I don't think the critical section will panic at all. As such the result is identical the `Unlock` can be moved to end. Though, as long as we're also aware of the fact that any newly introduced code prone to `panic` will introduce a potential bug in future.

thoughts?